### ISSUE TITLE
/UniTalker/unitalker_data_release_V1/D5_unitalker_faceforensics++/id_template.npy

### ISSUE THREAD

**Issue by duan348733684** [X-niper/UniTalker#11]
I opened this file, and its shape is 719x413. 
What does 413 mean?
From the code：
     self.FLAME_CONSTS = {
            'betas': 400,
            'rotation': 6,
            'jaw': 3,
            'eyeballs': 0,
            'neck': 0,
            'translation': 3,
            'scale': 1,
        }
 I see that betas is 400.  Does it mean 3DMM expressions + shape:  300 + 100 or shape+ expressions: 100 + 300 ?

**Comment by X-niper** [comment]
> I opened this file, and its shape is 719x413. What does 413 mean? From the code： self.FLAME_CONSTS = { 'betas': 400, 'rotation': 6, 'jaw': 3, 'eyeballs': 0, 'neck': 0, 'translation': 3, 'scale': 1, } I see that betas is 400. Does it mean 3DMM expressions + shape: 300 + 100 or shape+ expressions: 100 + 300 ?

Hi,  the betas array includes shape 300, expression 100 in the order of 300 + 100.  Adding betas,400, rotation 6, jaw 3, translation 3 and scale 1 we get the flame parameters of 413 length. 

It can be transferred into vertices using our code.

**Comment by duan348733684** [comment]
> > I opened this file, and its shape is 719x413. What does 413 mean? From the code： self.FLAME_CONSTS = { 'betas': 400, 'rotation': 6, 'jaw': 3, 'eyeballs': 0, 'neck': 0, 'translation': 3, 'scale': 1, } I see that betas is 400. Does it mean 3DMM expressions + shape: 300 + 100 or shape+ expressions: 100 + 300 ?
> 
> Hi, the betas array includes shape 300, expression 100 in the order of 300 + 100. Adding betas,400, rotation 6, jaw 3, translation 3 and scale 1 we get the flame parameters of 413 length.
> 
> It can be transferred into vertices using our code.


Thank you for your reply.
What does rotation=6 mean above?

**Comment by gzchenjiajun** [comment]
So I need to get the first 50 expressions. Is this how I get them?


`

            out_motion, out_pca, gt_pca = unitalker_models(
                audio_data,
                template,
                face_motion=None,
                style_idx=condition_id,
                annot_type=annot_type,
                fps=fps,
            )

	   out_list.append(out_motion.detach().cpu().numpy().squeeze(0))


bs_list = merge_out_list(out_list, fps)
expresion_list = bs_list[:, 300:350]
`
